projects/plugins/jetpack/extensions/blocks/button/edit.js

```diff
@@ -8,21 +8,16 @@ import {
 import { compose } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
 import clsx from 'clsx';
-import useWidth from '../../shared/use-width';
-import applyFallbackStyles from './apply-fallback-styles';
 import { IS_GRADIENT_AVAILABLE } from './constants';
 import ButtonControls from './controls';
 import usePassthroughAttributes from './use-passthrough-attributes';
 import './editor.scss';
 
 export function ButtonEdit( props ) {
-	const { attributes, backgroundColor, className, clientId, context, setAttributes, textColor } =
-		props;
+	const { attributes, backgroundColor, className, clientId, setAttributes, textColor } = props;
 	const { borderRadius, element, placeholder, text, width, fontSize } = attributes;
-	const isWidthSetOnParentBlock = 'jetpack/parentBlockWidth' in context;
 
 	usePassthroughAttributes( { attributes, clientId, setAttributes } );
-	useWidth( { attributes, disableEffects: isWidthSetOnParentBlock, setAttributes } );
```

Comment:
Calling this hook results in resetting the button width when it is aligned left or right. I don't get the rationale behind that decision. In the current UI, from a user standpoint, this doesn't seem like something we'd want.

Reply:
I've been testing and I also couldn't see the use case that the hook caters to. I saw this was implemented in #25394 by @mmtr, maybe Miguel remembers.

If there's no apparent current use cases, I wonder if it can be removed completely from the codebase. It's also used by Payment Button, but that has no alignment tool on the toolbar, so perhaps it's not needed.